tagblock.go

```diff
@@ -0,0 +1,99 @@
+package nmea
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+var (
+	// tagBlockRegexp matches nmea tag blocks
+	tagBlockRegexp = regexp.MustCompile(`^(.*)\\(\S+)\\(.*)`)
+)
+
+// TagBlock struct
+type TagBlock struct {
+	Head         string // *
+	Time         int64  // TypeUnixTime unix timestamp (unit is likely to be s, but might be ms, YMMV), parameter: -c
+	RelativeTime int64  // TypeRelativeTime relative time, parameter: -r
+	Destination  string // TypeDestinationID destination identification 15 char max, parameter: -d
+	Grouping     string // TypeGrouping sentence grouping, parameter: -g
+	LineCount    int64  // TypeLineCount line count, parameter: -n
+	Source       string // TypeSourceID source identification 15 char max, parameter: -s
+	Text         string // TypeTextString valid character string, parameter -t
+}
+
+func parseInt64(raw string) (int64, error) {
+	i, err := strconv.ParseInt(raw, 10, 64)
+	if err != nil {
+		return 0, fmt.Errorf("nmea: tagblock unable to parse uint64 [%s]", raw)
+	}
+	return i, nil
+}
+
+// parseTagBlock adds support for tagblocks
+// https://rietman.wordpress.com/2016/09/17/nemastudio-now-supports-the-nmea-0183-tag-block/
```

Comment:
This links to an nmeastudio block post. Is there a better reference we can use?

Reply:
My choice would be https://gpsd.gitlab.io/gpsd/AIVDM.html#_nmea_tag_blocks

Problem is that the NEMA 4.10 is not open/free...

What do you think?

Reply:
Yeah I couldn't find anything better either.

Reply:
@icholy done: b80f98f